src/opera_disp_tms/generate_sw_disp_tile.py

```diff
@@ -0,0 +1,233 @@
+    end_date = datetime.strftime(end_date, date_fmt)
+    name = '_'.join(['SW_CUMUL_DISP', begin_date, end_date, *parts])
+    return name
+
+
+def create_sw_disp_tile(metadata_path: Path, begin_date: datetime, end_date: datetime) -> Path:
+    """Create a short wavelength cumulative displacement tile
+
+    Args:
+        metadata_path: Path to the metadata GeoTiff file
+        begin_date: The beginning of the date range to generate the tile for
+        end_date: The end of the date range to generate the tile for
+
+    Returns:
+        Path to the generated tile
+    """
+    if not metadata_path.exists():
+        raise FileNotFoundError(f'{metadata_path} does not exist')
+    if begin_date > end_date:
+        raise ValueError('Begin date must be before end date')
+
+    product_name = create_product_name(metadata_path.name, begin_date, end_date)
+    product_path = metadata_path.parent / product_name
+
+    frame_map = utils.get_raster_array(metadata_path)
+
+    create_blank_copy_tile(metadata_path, product_path)
+    ds = gdal.Open(str(product_path), gdal.GA_Update)
+    band = ds.GetRasterBand(1)
+    sw_cumul_disp = band.ReadAsArray()
+
+    transfrom = Affine.from_gdal(*gdal.Open(str(metadata_path)).GetGeoTransform())
+    frames = frames_from_metadata(metadata_path)
+    frame_ids = [x.frame for x in frames]
+    needed_granules = find_needed_granules(frame_ids, begin_date, end_date)
+    secondary_dates = {}
+    for granule in needed_granules:
+        granule = open_opera_disp_granule(granule.s3_uri, 'short_wavelength_displacement')
+        granule_frame = [x for x in frames if x.frame == granule.attrs['frame']][0]
+        granule = update_spatiotemporal_reference(granule, granule_frame)
+        granule = granule.rio.reproject('EPSG:3857', transform=transfrom, shape=frame_map.shape)
+        frame_locations = frame_map == granule.attrs['frame']
+        sw_cumul_disp[frame_locations] = granule.data[frame_locations].astype(float)
+        secondary_date = datetime.strftime(granule.attrs['secondary_date'], DATE_FORMAT)
+        secondary_dates[f'FRAME_{granule_frame.frame}_SEC_TIME'] = secondary_date
+
+    band.WriteArray(sw_cumul_disp)
+    metadata = ds.GetMetadata()
+    metadata.update(secondary_dates)
+    ds.SetMetadata(metadata)
+    ds.FlushCache()
+    ds = None
+    return product_path
+
+
+def main():
+    """CLI entrpypoint
+    Example:
+    generate_sw_disp_tile METADATA_ASCENDING_N41W125_N42W124.tif --begin-date 20170901 --end-date 20171231
+    """
+    parser = argparse.ArgumentParser(description='Create a short wavelength cumulative displacement tile')
+    parser.add_argument('metadata_path', type=str, help='Path to the metadata GeoTiff file')
+    parser.add_argument('--begin-date', type=str, help='Start of date range to generate tile for in format: %Y%m%d')
+    parser.add_argument('--end-date', type=str, help='End of date range to generate tile for in format: %Y%m%d')
```

Comment:
Immediately throws on line 226 if begin date or end date are not provided, so these should be required positional parameters instead of optional `--` parameters:

```suggestion
    parser.add_argument('begin_date', type=str, help='Start of date range to generate tile for in format: %Y%m%d')
    parser.add_argument('end_date', type=str, help='End of date range to generate tile for in format: %Y%m%d')
```